### PR TITLE
Feature/text widget style options

### LINKED ISSUE #1810: [Feature] Introduce font sizes+alignment options in the text widget
Could we introduce these font sizes+alignment options in the text widget too? They feel like related issues which should be tackled all together. It'll also mean that the text widget doesn't appear inconsistent compared to the table widget options. 
Current list of properties:
Screen Shot 2020-11-18 at 1 16 54 PM

_Originally posted by @areyabhishek in https://github.com/appsmithorg/appsmith/issues/1771#issuecomment-729502237_

### PR DESCRIPTION
**Pull request by vicky-primathon**
## Description
Added styling options in text widget.

Fixes #1810 

## Type of change
- New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/components/designSystems/blueprint/TextComponent.tsx [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { TextStyle, TextAlign } from "widgets/TextWidget";
 import Interweave from "interweave";
 import { UrlMatcher, EmailMatcher } from "interweave-autolink";
 import { labelStyle } from "constants/DefaultTheme";
+import { FontStyleTypes } from "../appsmith/TableComponent/Constants";
```
Comment (riodeuno) [inline app/client/src/components/designSystems/blueprint/TextComponent.tsx:9]: Let's move this to `WidgetConstants`

File: app/client/src/widgets/TextWidget.tsx [hunk 2]
```diff
@@ -77,21 +90,82 @@ class TextWidget extends BaseWidget<TextWidgetProps, WidgetState> {
...
+          // {
+          //   propertyName: "textSize",
+          //   label: "Text Size",
+          //   controlType: "DROP_DOWN",
+          //   options: [
+          //     {
+          //       label: "Heading 1",
+          //       value: "HEADING1",
+          //       subText: "24px",
+          //       icon: "HEADING_ONE",
```
Comment (riodeuno) [inline app/client/src/widgets/TextWidget.tsx:102]: Let's remove these commented lines of code.

File: app/client/src/widgets/TextWidget.tsx [hunk 3]
```diff
@@ -140,14 +217,20 @@ class TextWidget extends BaseWidget<TextWidgetProps, WidgetState> {
...
-export type TextStyle = "BODY" | "HEADING" | "LABEL" | "SUB_TEXT";
+export type TextStyle = "BODY" | "HEADING" | "LABEL";
 export type TextAlign = "LEFT" | "CENTER" | "RIGHT" | "JUSTIFY";
 
-export interface TextWidgetProps extends WidgetProps {
-  text?: string;
+export interface TextStyles {
+  backgroundColor?: string;
+  textColor?: string;
+  fontStyle?: string;
```
Comment (riodeuno) [inline app/client/src/widgets/TextWidget.tsx:226]: Let's make sure we have proper default values here: (WidgetConfigResponse)
Screenshot 2021-03-11 at 5 39 38 AM

Text color & font style values.

**Review by riodeuno — CHANGES_REQUESTED** [review]

**Comment by riodeuno** [comment]
- One of the tests are failing
- We need a migration for old "Text Style" property. Along with that we need to use the font sizes property and deprecate "Text Style". The old "Text Style" properties can all be defined using a combination of "Font Style" and "Font Size" properties.
Screenshot 2021-03-11 at 5 39 55 AM

File: app/client/src/widgets/TextWidget.tsx (hunk 3 above)
Comment (vicky-primathon) [reply app/client/src/widgets/TextWidget.tsx:188]: Added default color, font style and font size.

File: app/client/src/components/propertyControls/DropDownControl.tsx [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import React from "react";
 import BaseControl, { ControlProps } from "./BaseControl";
 import { StyledDropDown, StyledDropDownContainer } from "./StyledControls";
```
Comment (riodeuno) [inline app/client/src/components/propertyControls/DropDownControl.tsx:3]: Please verify if this gives us an eslint warning about unused vars

File: app/client/src/utils/migrations/TextWidgetReplaceTextStyle.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import { ContainerWidgetProps } from "widgets/ContainerWidget";
+import { WidgetProps } from "widgets/BaseWidget";
+import { cloneDeep } from "lodash";
+import { WidgetTypes } from "constants/WidgetConstants";
+
+export const migrateTextStyleFromTextWidget = (
+  currentDSL: ContainerWidgetProps<WidgetProps>,
+): ContainerWidgetProps<WidgetProps> => {
+  currentDSL.children = currentDSL.children?.map((_child: WidgetProps) => {
+    let child = cloneDeep(_child);
```
Comment (riodeuno) [inline app/client/src/utils/migrations/TextWidgetReplaceTextStyle.ts:10]: Let's add unit tests for this.

**Review by riodeuno — CHANGES_REQUESTED** [review]

**Review by riodeuno — DISMISSED** [review]

File: app/client/src/constants/WidgetConstants.tsx [hunk 6]
```diff
@@ -115,3 +115,28 @@ export const MAIN_CONTAINER_WIDGET_NAME = "MainContainer";
 export const WIDGET_DELETE_UNDO_TIMEOUT = 7000;
 
 export const DEFAULT_CENTER = { lat: -34.397, lng: 150.644 };
+
+export enum FontStyleTypes {
```
Comment (riodeuno) [inline app/client/src/constants/WidgetConstants.tsx:119]: We need to write tests for functions in which we're referring to these constants.

**Review by riodeuno — APPROVED** [review]

**Review by riodeuno — APPROVED** [review]

**Review by riodeuno — APPROVED** [review]

**Review by riodeuno — APPROVED** [review]

**Pull request merged**